sycl/include/sycl/ext/intel/esimd/detail/atomic_intrin.hpp

```diff
@@ -22,4 +41,82 @@ template <typename Ty> Ty atomic_add_fetch(Ty *ptr, Ty val) {
 #endif
 }
 
+template <typename Ty> Ty atomic_sub_fetch(Ty *ptr, Ty val) {
+#ifdef _WIN32
+  // TODO: Windows will be supported soon
+  __ESIMD_UNSUPPORTED_ON_HOST;
+#else
+  return __atomic_sub_fetch(ptr, val, __ATOMIC_RELAXED);
```

Comment:
I'd suggest to use __ATOMIC_SEQ_CST every where to be on the safe side, unless we can prove relaxed is OK here.

Reply:
Done.